tools/src/wyvern/target/corewyvernIL/expression/MethodCall.java

```diff
@@ -316,8 +348,14 @@ public DefDeclType typeMethodDeclaration(TypeContext ctx, EffectAccumulator effe
                 return defDeclType;
             }
         }
-
         // Couldn't find an appropriate method declaration. Build up a nice error message.
+        methodDeclarationNotFound(ctx, declarationTypes, actualArgTypes, formalArgTypes, failureReason);
+        throw new RuntimeException("Tool error should be thrown");
```

Comment:
Could you please change the `RuntimeException`'s message to something else, perhaps "Impossible," and add a comment saying that an exception/tool error will be thrown from `methodDeclarationNotFound()` right above this line?